terratorch/datasets/generic_pixel_wise_dataset.py

```diff
@@ -120,19 +121,38 @@ def __init__(
             )
         self.rgb_indices = [0, 1, 2] if rgb_indices is None else rgb_indices
 
-        self.dataset_bands = dataset_bands
-        self.output_bands = output_bands
+        bands_by_interval = (self._bands_defined_by_interval(bands_list=dataset_bands) and
+                             self._bands_defined_by_interval(bands_list=output_bands))
+
+        # If the bands are defined by sub-intervals or not.
+        if bands_by_interval:
+            self.dataset_bands = self._generate_bands_intervals(dataset_bands)
+            self.output_bands = self._generate_bands_intervals(output_bands)
+        else:
+            self.dataset_bands = dataset_bands
+            self.output_bands = output_bands
+            
+            bands_type = self._bands_as_int_or_str(dataset_bands, output_bands)
+
+            if bands_type == str:
+                UserWarning("When the bands are defined as str, guarantee your input files"+ 
+                                  "are organized by band and all have its specific name.")
```

Comment:
I don't think this is worth a warning, the user could be doing everything correctly but would receive a warning every time they run their code. We should instead make it clear in the documentation / docstring exactly what each case expects (str vs range and int)

Reply:
303bfe89a22a20cbe65de90fbf4d701aff7df25f